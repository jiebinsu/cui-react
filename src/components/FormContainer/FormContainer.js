import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";

const FormContainer = ({ className, children, ...props }) => (
  <form className={classNames("coop-form", [className])} {...props}>
    {children}
  </form>
);

FormContainer.defaultProps = {
  className: null,
  children: null,
};

FormContainer.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};

export default FormContainer;
