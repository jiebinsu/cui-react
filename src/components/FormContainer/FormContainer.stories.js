import React from "react";
import { action } from "@storybook/addon-actions";
import FormContainer from "./FormContainer";

export default {
  title: "Base/FormContainer",
  component: FormContainer,
};

export const Basic = (args) => <FormContainer {...args} />;
Basic.args = {};

export const CustomProps = () => (
  <FormContainer
    id="custom"
    className="customClass"
    onClick={action("clicked")}
  >
    Props will be propagated to the form element
  </FormContainer>
);
