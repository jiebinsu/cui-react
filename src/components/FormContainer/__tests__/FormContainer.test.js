import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import FormContainer from "../index";

describe("FormContainer", () => {
  it("should render default FormContainer", () => {
    const wrapper = mount(<FormContainer />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<FormContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should work with props", () => {
    const onClickMock = jest.fn();
    const onSubmitMock = jest.fn();
    const WithProps = () => (
      <FormContainer
        id="test"
        className="newClass"
        onClick={() => onClickMock}
        onSubmit={() => onSubmitMock}
      >
        Form Body
      </FormContainer>
    );

    const wrapper = mount(<WithProps />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer.create(<WithProps />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
