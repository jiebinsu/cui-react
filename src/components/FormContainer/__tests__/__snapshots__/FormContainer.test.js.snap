// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormContainer should render default FormContainer 1`] = `
<form
  className="coop-form"
/>
`;

exports[`FormContainer should work with props 1`] = `
<form
  className="coop-form newClass"
  id="test"
  onClick={[Function]}
  onSubmit={[Function]}
>
  Form Body
</form>
`;
